spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,150 @@
+    implements AnalyzeTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+  private final Table table;
+  private Set<String> columnsToBeAnalyzed;
+  private Set<String> statsToBecollected =
+      Sets.newHashSet(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1);
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    String desc = String.format("Analyzing table %s", table.name());
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    LOG.info("Starting the analysis of {}", table.name());
+    validateColumns();
+    List<AnalysisResult> analysisResults =
+        statsToBecollected.stream()
+            .map(
+                statsName -> {
+                  switch (statsName) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      return generateNDVAndCommit();
+                    default:
+                      return ImmutableAnalyzeTable.AnalysisResult.builder()
+                          .statsName(statsName)
+                          .statsCollected(false)
+                          .addAllErrors(Lists.newArrayList("Stats type not supported"))
+                          .build();
+                  }
+                })
+            .collect(Collectors.toList());
+    return ImmutableAnalyzeTable.Result.builder().analysisResults(analysisResults).build();
+  }
+
+  private void validateColumns() {
+    validateEmptyColumns();
+    validateTypes();
+  }
+
+  private void validateEmptyColumns() {
+    if (columnsToBeAnalyzed == null || columnsToBeAnalyzed.isEmpty()) {
+      throw new ValidationException("No columns to analyze for the table", table.name());
+    }
+  }
+
+  private void validateTypes() {
+    columnsToBeAnalyzed.forEach(
+        columnName -> {
+          Types.NestedField field = table.schema().findField(columnName);
+          if (field == null) {
+            throw new ValidationException("No column with %s name in the table", columnName);
+          }
+          Type type = field.type();
+          if (type.isListType() || type.isMapType() || type.isNestedType() || type.isStructType()) {
+            throw new ValidationException(
+                "Analysis not supported on column %s of type %s", columnName, type.typeId());
+          }
+        });
+  }
+
+  private AnalysisResult generateNDVAndCommit() {
+    try {
+      StatisticsFile statisticsFile =
+          NDVSketchGenerator.generateNDV(
+              spark(), table, columnsToBeAnalyzed.toArray(new String[0]));
+      table
+          .updateStatistics()
+          .setStatistics(table.currentSnapshot().snapshotId(), statisticsFile)
```

Comment:
what of table's current snapshot has modified concurrently by another client between like 117 to line 120?

Reply:
this is a good question. we should do https://github.com/apache/iceberg/issues/6442